include/cell.h

```diff
@@ -222,6 +224,29 @@ class Cell {
                                     unsigned phase, double pvolume,
                                     const Eigen::Matrix<double, 6, 1>& pstress);
 
+  //! Assign velocity constraint
+  //! Directions can take values between 0 and Dim-1
+  //! \param[in] face_id Face of cell of velocity constraint
+  //! \param[in] dir Direction of velocity constraint
+  //! \param[in] velocity Applied velocity constraint
+  bool assign_cell_velocity_constraint(unsigned face_id, unsigned dir,
+                                       double velocity);
+
+  //! Apply velocity constraints
+  void apply_cell_velocity_constraints();
+
+  //! Compute normal vector
+  void compute_normals();
+
+  //! Return number of normal vectors
+  //! \retval number of normal vectors
+  unsigned nnormal() const { return face_normals_.size(); }
+
+  //! Return unit normal vector, positive pointing outside of the element
+  //! \param[in] face_id of constraint
+  //! \retval unit normal vector
+  Eigen::VectorXd normal(unsigned face_id);
```

Comment:
I need to check where this is used?

Reply:
We need to call the normal later to calculate euler angles before computing rotation matrix

Reply:
Should this be public?